src/runtime/audience-action-flow.js

```diff
@@ -91,17 +91,25 @@ export class AudienceActionFlow {
     /** @private @const {?./client-config-manager.ClientConfigManager} */
     this.clientConfigManager_ = deps.clientConfigManager();
 
-    /** @private @const {!ActivityIframeView} */
-    this.activityIframeView_ = new ActivityIframeView(
-      deps.win(),
-      deps.activities(),
-      this.getUrl_(deps.pageConfig(), deps.win()),
-      feArgs({
-        'supportsEventManager': true,
-        'productType': this.productType_,
-      }),
-      /* shouldFadeBody */ true
-    );
+    /** @private @const {!Promise<?ActivityIframeView>} */
+    this.activityIframeViewPromise_ = deps
+      .storage()
+      .get(Constants.USER_TOKEN, true)
+      .then((token) => {
```

Comment:
storage().get() can return NULL which will result in sud=NULL in your URL - is the your intent and can the backend cope with that?

[we can't guarantee that some extension hasn't trashed storage or if we are in an iframe on safari storage doesn't persist across sessions]

Reply:
In line 168, if storage().get() returns null, we will set empty string to the sut param.